### DISCUSSION TITLE
Unknown behavior: SynctaxError: Unexpected number at new Function (<anonymous>)

### DISCUSSION THREAD

**Discussion by saeedhbi** [mikro-orm/mikro-orm#5388] [General]
### Describe the bug

I'm using MikroOrm with MongoDB in a pure typescript project. The problem is when I'm creating a new instance of an entity, it throws an Unhandled promise rejection as `SynctaxError: Unexpected number at new Function (<anonymous>)`.

**Unfortunately, I can't create a reproduction example since this is happening in my machine and I can't find the problem.**

I'm not passing any `number` value to a property and it seems there is a problem regarding context.

The Stack is this:

```'SyntaxError: Unexpected number at new Function (<anonymous>)'
[
    {
      functionName: 'at Utils.createFunction (/www/node_modules/.pnpm/@mikro-orm+core@5.7.11_@mikro-orm+migrations-mongodb@5.7.11_@mikro-orm+migrations@5.7.11_@mik_hjftnkmntakp6rrzqlupf3vsey/node_modules/@mikro-orm/core/utils/Utils.js:790:20)',
      fileName: 'at Utils.createFunction (//www/node_modules/.pnpm/',
      lineNumber: 'mikro-orm+core@5.7.11_@mikro-orm+migrations-mongodb@5.7.11_@mikro-orm+migrations@5.7.11_@mik_hjftnkmntakp6rrzqlupf3vsey/node_modules/@mikro-orm/core/utils/Utils.js',
      columnNumber: '790'
    },
    {
      functionName: 'at ObjectHydrator.getEntityHydrator (/www/node_modules/.pnpm/@mikro-orm+core@5.7.11_@mikro-orm+migrations-mongodb@5.7.11_@mikro-orm+migrations@5.7.11_@mik_hjftnkmntakp6rrzqlupf3vsey/node_modules/@mikro-orm/core/hydration/ObjectHydrator.js:284:40)',
      fileName: 'at ObjectHydrator.getEntityHydrator (/www/node_modules/.pnpm/',
      lineNumber: 'mikro-orm+core@5.7.11_@mikro-orm+migrations-mongodb@5.7.11_@mikro-orm+migrations@5.7.11_@mik_hjftnkmntakp6rrzqlupf3vsey/node_modules/@mikro-orm/core/hydration/ObjectHydrator.js',
      columnNumber: '284'
    },
...]
```

### Reproduction

-

### What driver are you using?

@mikro-orm/mongodb

### MikroORM version

5.7.11

### Node.js version

v20.1.0

### Operating system

Ubuntu

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> 5.7.11

The first thing you should try is upgrading, this is very old. Ideally use latest v6 version.

**Reply by saeedhbi** [reply]
Thanks for the update. I will consider upgrading, but if the problem is clear, please let me know. Thanks.

**Reply by saeedhbi** [reply]
I upgraded to the latest version and the error exists as before.

**Reply by B4nan** [reply]
How do your entities look like?

**Reply by saeedhbi** [reply]
This is an example of entity I do have problem:

```
@Entity({ repository: () => ProductRepository, tableName: "products" })
@Index({ properties: ["provider"] })
@Index({ properties: ["providerProductId"] })
@Index({ properties: ["lang"] })
@Index({ properties: ["state"] })
@Index({ properties: ["labels"] })
export class ProductEntity extends BaseMongoEntity {
	[EntityRepositoryType]?: ProductRepository;

	@Index({ options: { "details.lang": "2dsphere" } })
	@Index({ options: { "details.title": "2dsphere" } })
	@Index({ options: { "details.slug": "2dsphere" } })
	@Index({ options: { "details.price.$**": "3dsphere" } })
	@Index({ options: { "details.information.$**": "3dsphere" } })
	@Embedded(() => ProductDetailsEntity, { array: true, object: true, nullable: true })
	details: ProductDetailsEntity[] = [];

	@Enum({ type: EnumType, nullable: true, items: SUPPORTED_PROVIDERS })
	provider: TProviders;

	@Property({ nullable: false })
	providerProductId!: string;

	@Enum({ type: EnumType, nullable: false, items: SUPPORTED_PRODUCT_STATE })
	state: TProductState;

	@Enum({ type: EnumType, items: SUPPORTED_LANGS })
	lang: TLangs = LANGS.EN;

	@Enum({ type: EnumArrayType, nullable: true, items: SUPPORTED_LABELS })
	labels: TLabels[] = [];

	@Index({ options: { "tags.$**": "2dsphere" } })
	@Property({ nullable: true })
	tags: TagEntity[] = [];

	@Index({ options: { "files.filename": "2dsphere" } })
	@Embedded(() => [FileImageEntity, FileVideoEntity], { array: true, object: true, nullable: true })
	files: (FileImageEntity | FileVideoEntity)[] = [];

	@Index({ options: { "analytics.$**": "2dsphere" } })
	@Embedded(() => ProductAnalyticsEntity, { nullable: true, object: true })
	analytics: ProductAnalyticsEntity;

	@Property({ nullable: true })
	relatedCapacityIds?: string[];

	@Property({ nullable: true })
	relatedProductIds?: string[];
}

```

On creation of entity, I'm using like this:

`orm.getRepository(ProductEntity).create(product)`

**Reply by B4nan** [reply]
Try to simplify till it works. Also errors from compiled functions should be always preceded with the function implementation, so look for logs above this.

My guess is you have some property mapped to a column that start with a number or something like that.

**Reply by saeedhbi** [reply]
Yes. The problem was with one of the Embedded entities. Thanks.
